### ISSUE TITLE
[Bug]: CV11 triggers on aggregate functions

### ISSUE THREAD

**Issue by asqarslanov** [quarylabs/sqruff#2042]
### What happened?

The linter emits false positive C11 (`convention.casting_style`) warnings on aggregate functions.

`.sqruff` (all rules enabled):

- ```ini
  [sqruff]
  rules = all
  ```

SQL query:

- ```sql
  SELECT
      a::int, -- shorthand casting
      count(b) AS c -- incorrectly identified as another style of casting
  FROM
      foo;
  ```

### Version

0.29.3

### Checked in Playground

- [x] I have checked this bug in the [playground](https://playground.quary.dev/?secondary=Format).
